### PR TITLE
TECH-565: Cucumber test for GET /applications/{fileId}

### PR DESCRIPTION
**Pull request by mdulko-soldevelo**
Ticket: https://govstack-global.atlassian.net/browse/TECH-565

### CODE REVIEW

File: test/openAPI/features/applicant_applications_fileId.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@method=POST @endpoint=/applications/{fileId}
+Feature: Returns application file
```
Comment (KarolinaKopacz) [inline test/openAPI/features/applicant_applications_fileId.feature]:
```suggestion
Feature: The API endpoint used to get application file by ID
```

File: test/openAPI/features/applicant_applications_fileId.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@method=POST @endpoint=/applications/{fileId}
```
Comment (KarolinaKopacz) [inline test/openAPI/features/applicant_applications_fileId.feature]: Please change the file name `applicant_applications_fieldid` or `applicant_applications_field_id`

File: test/openAPI/features/support/applicant_applications_fileId.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+const chai = require('chai');
```
Comment (KarolinaKopacz) [inline test/openAPI/features/support/applicant_applications_fileId.js]: Please change this file name too

File: test/openAPI/features/support/helpers/helpers.js [hunk 4]
```diff
@@ -45,6 +45,49 @@ module.exports = {
       },
     ],
   },
+  //applicant_applications_fileId
+  applicationFileEndpoint: 'applications/{fileId}',
+  applicationFileResponseSchema: {
+    type: 'object',
+    properties: {
+      fileId: { type: 'string' },
+      registered: { type: 'string' },
```
Comment (KarolinaKopacz) [inline test/openAPI/features/support/helpers/helpers.js]: In $date-time format, example: 2000-10-23T00:00:00.000Z

File: test/openAPI/features/support/helpers/helpers.js [hunk 5]
```diff
@@ -45,6 +45,49 @@ module.exports = {
...
+          title: { type: 'string' }
+        }
+      },
+      ended: { type: 'boolean' },
+      applicationData: {
+        type: 'object',
+        properties: {
+          applicationName: { type: 'string' },
+          applicantId: { type: 'string' },
+          created: { type: 'string' },
```
Comment (KarolinaKopacz) [inline test/openAPI/features/support/helpers/helpers.js]: In $date-time format, example: 2000-10-23T00:00:00.000Z

File: test/openAPI/features/support/helpers/helpers.js [hunk 6]
```diff
@@ -45,6 +45,49 @@ module.exports = {
...
+          formData: {
+            type: 'object',
+            properties: {
+              data: { type: 'object' },
+              eFormId: { type: 'string' },
+            },
+          },
+          documentUrls: { type: "array" }
```
Comment (KarolinaKopacz) [inline test/openAPI/features/support/helpers/helpers.js]: array of strings
also `documentUrls` is used in another response schema so good to add it in shares section

File: test/openAPI/features/support/helpers/helpers.js [hunk 7]
```diff
@@ -45,6 +45,49 @@ module.exports = {
...
+              'APPROVED',
+              'SENDBACK',
+              'REJECTED'
+            ]
+          },
+          title: { type: 'string' }
+        }
+      },
+      ended: { type: 'boolean' },
+      applicationData: {
```
Comment (KarolinaKopacz) [inline test/openAPI/features/support/helpers/helpers.js]: This part of the response schema is used several times in the API spec, so good to add it in shares section and name it `applicationFileSubmit`

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: test/openAPI/features/applicant_applications_fileId.feature (hunk 2 above)
Comment (mgrochalskisoldevelo) [reply test/openAPI/features/applicant_applications_fileId.feature]: I agree, but fileid -> not fieldid

File: test/openAPI/features/support/applicant_applications_fileId.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  fileId =>
+    specApplicantApplicationsFileId
+      .get(baseUrl)
+      .withHeaders(header.key, header.value)
+      .withPathParams('fileId', fileId)
+);
+
+Then(
+  /^The response from the \/applications\/\{fileId\} endpoint is received$/,
+  () => specApplicantApplicationsFileId.toss()
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/applicant_applications_fileId.js]: async () => await specApplicantApplicationsFileId.toss()

File: test/openAPI/features/support/helpers/helpers.js [hunk 9]
```diff
@@ -45,6 +45,49 @@ module.exports = {
       },
     ],
   },
+  //applicant_applications_fileId
+  applicationFileEndpoint: 'applications/{fileId}',
+  applicationFileResponseSchema: {
```
Comment (mgrochalskisoldevelo) [inline test/openAPI/features/support/helpers/helpers.js:50]: `serviceId` and `applicantId` should be required in the response schema

**Review by mgrochalskisoldevelo — CHANGES_REQUESTED** [review]

File: test/openAPI/features/support/helpers/helpers.js (hunk 4 above)
Comment (mdulko-soldevelo) [reply test/openAPI/features/support/helpers/helpers.js]: What do you mean? Should be there sone other type?
updated the rest just have no idea about this one

Comment (KarolinaKopacz) [reply test/openAPI/features/support/helpers/helpers.js]:
```
{
      "type": "string",
      "format": "date-time"
 }
```
https://opis.io/json-schema/2.x/formats.html
In the API spec there is information that this field should have date-time format

**Review by KarolinaKopacz — APPROVED** [review]

**Review by ingmarvali — APPROVED** [review]

**Pull request merged**